cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -169,6 +169,19 @@ inline compute::Expression UseBoringRefs(const compute::Expression& expr) {
   return compute::Expression{std::move(modified_call)};
 }
 
+Status CheckTable(std::shared_ptr<Buffer>& buf,
+                  const ConversionOptions& conversion_options = {}) {
+  std::shared_ptr<ExtensionIdRegistry> sp_ext_id_reg = MakeExtensionIdRegistry();
+  ExtensionIdRegistry* ext_id_reg = sp_ext_id_reg.get();
+  ExtensionSet ext_set(ext_id_reg);
+  ARROW_ASSIGN_OR_RAISE(auto sink_decls, DeserializePlans(
```

Comment:
There is now an `arrow::engine::DeserializePlan` which returns a single `Declaration` that you can use here.

Reply:
Removed.